pkg/registries/registries.go

```diff
@@ -3,17 +3,53 @@ package registries
 import (
 	"fmt"
 	"sort"
+	"strings"
 
 	config "github.com/openshift/api/config/v1"
 )
 
 // mirror represents a mirrored image repo entry in a registry configuration file
 type mirror struct {
+	// host is the mirror image location. Can include the registry hostname/IP address, port, and namespace path
 	host string
 	// resolveTags indicates to the container runtime if this mirror is allowed to resolve an image tag into a digest
 	resolveTags bool
 }
 
+// newMirror constructs a new mirror object with proper host name structure to be used in containerd registry config
+func newMirror(sourceImageLocation, mirrorImageLocation string, resolveTags bool) mirror {
+	mirrorHost := ""
+	// containerd appends any shared namespaces between source and mirror locations to the mirror's host entry in the
+	// registry config file to construct the full mirror image location
+	if sourceImageLocation != mirrorImageLocation {
+		// truncate the mirror to drop any shared namespaces since containerd automatically appends them on image pull
+		mirrorHost = dropCommonSuffix(sourceImageLocation, mirrorImageLocation)
+	} else {
+		// special case if source and mirror are the same. Do not drop the host repo name to avoid an empty host entry
+		mirrorHost = extractHostname(mirrorImageLocation)
+	}
+	return mirror{host: mirrorHost, resolveTags: resolveTags}
+}
+
+// dropCommonSuffix drops the common suffix from the second repo, returning only the unique leading URL and namespaces
+func dropCommonSuffix(source, mirror string) string {
```

Comment:
Consider naming this as `extractMirrorURL`

dropping the common suffix is just the "how" it achieved the goal: returning only the unique leading URL and namespaces of the mirror